### ISSUE TITLE
List item inside annotation is rendering into a different annotation.

### ISSUE THREAD

**Issue by feasgal** [squidfunk/mkdocs-material#5496]
### Bug description

I have a sentence with two annotations in it. The first annotation includes an unordered list of three items. The second annotation is a single sentence.

The first annotation is rendering with the second unordered list item blank, and the second annotation's content is replaced by the text of that missing list item.

UPDATE:

- Upon further testing, this happens with ordered and unordered lists, and it happens whether I use markdown syntax for the lists, or insert raw HTML in the annotation.
- In the `<ol hidden>` that appears after the paragraph in the page source, the second `<li>` has not been stripped like the first one has.
- All of which leads me to believe it's something about the way the popup in the block with the `.annotate` class is rendered from the ordered list of annotations below it that gets hidden in the final page display.

### Related links

- [Annotations](https://squidfunk.github.io/mkdocs-material/reference/annotations/)

### Reproduction

[untitled folder.zip](https://github.com/squidfunk/mkdocs-material/files/11436656/untitled.folder.zip)

### Steps to reproduce

MkDocs version: 1.4.2
Material version: 9.1.1+insiders.4.32.2

Here's the markdown code:

```
A sentence with an annotation (1) and a second annotation (2). 
{ .annotate }

1.	The beginning of the first annotation, which includes a:
	
	-	list
	-	of
	-	items
	
	Plus another sentence.
	
2.	Here's the second annotation.
```

But the two annotations are rendering like this:
![image](https://github.com/squidfunk/mkdocs-material/assets/97999520/1950bde4-f04f-43d8-9ca7-e197a2419f72)

And this:
![image](https://github.com/squidfunk/mkdocs-material/assets/97999520/30abd727-3362-4cc4-8a99-31292e9b51db)

I'm baffled.

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting! Fixed in [`43c30522a`](https://github.com/squidfunk/mkdocs-material-insiders/commit/43c30522afe4bdbd89da17b0c3c314d17e621541). The problem was that the selector to find the respective lists was not specific enough, so the list inside the code annotation was mistaken for being the annotation content.

**Comment by squidfunk** [comment]
Released as part of [9.1.12+insiders-4.33.2](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.33.2).

**Comment by feasgal** [comment]
Wow, what a quick turn-around! It works as expected now. Thank you!
